systest/tests/distributed_post_verification_test.go

```diff
@@ -0,0 +1,298 @@
+	ptimesync := peersync.New(
+		host,
+		host,
+		peersync.WithLog(log.NewFromLog(logger.Named("peersync"))),
+		peersync.WithConfig(cfg.TIME.Peersync),
+	)
+	ptimesync.Start()
+	t.Cleanup(ptimesync.Stop)
+
+	require.NoError(t, host.Start())
+	t.Cleanup(func() { assert.NoError(t, host.Stop()) })
+
+	syncer := activation.NewMocksyncer(gomock.NewController(t))
+	syncer.EXPECT().RegisterForATXSynced().DoAndReturn(func() <-chan struct{} {
+		ch := make(chan struct{})
+		close(ch)
+		return ch
+	}).AnyTimes()
+
+	// 1. Initialize
+	postSetupMgr, err := activation.NewPostSetupManager(
+		signer.NodeID(),
+		cfg.POST,
+		logger.Named("post"),
+		datastore.NewCachedDB(sql.InMemory(), log.NewNop()),
+		cl.GoldenATX(),
+		syncer,
+		activation.NewMocknipostValidator(gomock.NewController(t)),
+	)
+	require.NoError(t, err)
+
+	postSupervisor, err := activation.NewPostSupervisor(
+		logger.Named("post-supervisor"),
+		cfg.POSTService,
+		cfg.POST,
+		cfg.SMESHING.ProvingOpts,
+		postSetupMgr,
+	)
+	require.NoError(t, err)
+	require.NoError(t, postSupervisor.Start(cfg.SMESHING.Opts))
+	t.Cleanup(func() { assert.NoError(t, postSupervisor.Stop(false)) })
+
+	// 2. create ATX with invalid POST labels
+	clock, err := timesync.NewClock(
+		timesync.WithLayerDuration(cfg.LayerDuration),
+		timesync.WithTickInterval(1*time.Second),
+		timesync.WithGenesisTime(cl.Genesis()),
+		timesync.WithLogger(log.NewFromLog(logger.Named("clock"))),
+	)
+	require.NoError(t, err)
+
+	grpcPostService := grpcserver.NewPostService(logger.Named("grpc-post-service"))
+	grpczap.SetGrpcLoggerV2(grpclog, logger.Named("grpc"))
+	grpcPrivateServer, err := grpcserver.NewPrivate(
+		logger.Named("grpc-server"),
+		cfg.API,
+		[]grpcserver.ServiceAPI{grpcPostService},
+	)
+	require.NoError(t, err)
+	require.NoError(t, grpcPrivateServer.Start())
+	t.Cleanup(func() { assert.NoError(t, grpcPrivateServer.Close()) })
+
+	nipostBuilder, err := activation.NewNIPostBuilder(
+		localsql.InMemory(),
+		activation.NewPoetDb(sql.InMemory(), log.NewNop()),
+		grpcPostService,
+		cfg.PoetServers,
+		logger.Named("nipostBuilder"),
+		signer,
+		cfg.POET,
+		clock,
+	)
+	require.NoError(t, err)
+
+	// 2.1. Create initial POST
+	var challenge *types.NIPostChallenge
+	for {
+		client, err := grpcPostService.Client(signer.NodeID())
+		if err != nil {
+			ctx.Log.Info("waiting for poet service to connect")
+			time.Sleep(time.Second)
+			continue
+		}
+		ctx.Log.Info("poet service to connected")
+		post, postInfo, err := client.Proof(ctx, shared.ZeroChallenge)
+		require.NoError(t, err)
+
+		challenge = &types.NIPostChallenge{
+			PrevATXID:      types.EmptyATXID,
+			PublishEpoch:   2,
+			PositioningATX: goldenATXID,
+			CommitmentATX:  &postInfo.CommitmentATX,
+			InitialPost:    post,
+		}
+		break
+	}
+
+	nipost, err := nipostBuilder.BuildNIPost(ctx, challenge)
+	require.NoError(t, err)
+
+	// 2.2 Create ATX with invalid POST
+	for i := range nipost.Post.Indices {
+		nipost.Post.Indices[i] += 1
+	}
```

Comment:
I think all of this shouldn't be necessary just to create an ATX with an invalid post? Can't the content of `nipost.Post` just be random?

Reply:
Alternatively you can use a remote smesher setup and add an additional init-container that overwrites `postdata_N.bin` with random bytes?

Reply:
Also consider that we will probably remove the `postSetupManager` from the node in the near future and replace it with `postcli` for initialization (directly or indirectly via smapp).

Reply:
The ATX needs a valid initial POST or it would be rejected before even checking its POST. Creating it already requires initializing data. It also needs valid poet proof. At this point it was easier to just create NiPOST normally and just corrupt POST proof indices.

Reply:
I'm still unsure about this test - I think there is an not insignificant chance that the corrupted ATX will still happen to be valid - we had colliding ATX IDs on fastnet / testnet before we added the NodeID to the ID calculation meaning different (valid) PoSTs had the same indices as solution.

Reply:
The test is configured to initialize 256 labels. Only roughly 12 of them are valid. The test picks indexes of 4 valid labels and modifies them. IMHO the chance to randomly pick 4 different labels is insignificant. This probability could be further decreased by increasing the number of initialized units (the "12 labels are valid" stays the same no matter of the total initialized size).

Reply:
I think this probability is `(11/255)**4 = 3.46e-6`.

Reply:
As discussed offline we can just add a check if the generated PoST is really invalid before we publish it to catch a collision early and quickly 🙂

Reply:
Done